### DISCUSSION TITLE
How could I add the neuron style uplink tree to the book template?

### DISCUSSION THREAD

**Discussion by Ninlives** [srid/emanote#260] [Q&A] [answered]
I'd like to have both side bar and uplink tree for navigation, so I'm trying the add the uplink tree to the default template but don't know what to do. I tried directly adding `<apply template="components/note-uptree" />` to `hooks/before-note.tpl` and apprently it does not work. Do you have a solution for this task?

**Comment by Ninlives** [comment, marked as answer]
I archieve this by modify the default template directly.
Is there a more blessed way to do this?

**Reply by srid** [reply]
@Ninlives Perhaps try using absolute path, i.e, use a `/` in front, viz:

```html
<apply template="/components/note-uptree" />
```

Or:

```html
<apply template="/templates/components/note-uptree" />
```

**Reply by Ninlives** [reply]
Oh, it works, and the solution seems to be obvious, I wonder why I didn't think about it at first..

**Comment by srid** [comment]
This is now possible. You can have _both_ sidebar and the uptree:

https://emanote.srid.ca/guide/html-template/neuron-layout

Furthermore, the sidebar now respects folgezettel links.
